lib/Entities/NewsletterEntity.php

```diff
@@ -257,6 +258,25 @@ public function getStatus() {
    */
   public function setStatus($status) {
     $this->status = $status;
+
+    $queues = $this->getUnfinishedQueues();
+    $newTaskStatus = null;
+    $typesWithActivation = [self::TYPE_NOTIFICATION, self::TYPE_WELCOME, self::TYPE_AUTOMATIC];
+
+    if (($status === self::STATUS_DRAFT) && in_array($this->type, $typesWithActivation)) {
+      $newTaskStatus = ScheduledTaskEntity::STATUS_PAUSED;
+    }
+    if (($status === self::STATUS_ACTIVE) && in_array($this->type, $typesWithActivation)) {
+      $newTaskStatus = ScheduledTaskEntity::STATUS_SCHEDULED;
+    }
+
+    if (!$newTaskStatus) return;
+    foreach ($queues as $queue) {
+      /** @var SendingQueueEntity $queue */
+      $task = $queue->getTask();
+      if ($task === null) continue;
+      $task->setStatus($newTaskStatus);
```

Comment:
I'm not sure whether we should always set the new status here. In the old models, we checked whether the task status was `paused` ([see here](https://github.com/mailpoet/mailpoet/blob/67824f9eb17abf4fa663c7964dedbdde301ba8b9/lib/Models/ScheduledTask.php#L75)) respectively `scheduled` ([see here](https://github.com/mailpoet/mailpoet/blob/67824f9eb17abf4fa663c7964dedbdde301ba8b9/lib/Models/ScheduledTask.php#L58)). I think it would be safer to stick with the previous behavior.